lib/appsignal/phoenix/channel.ex

```diff
@@ -94,11 +94,32 @@ if Appsignal.phoenix?() do
       <<"Elixir.", action::binary>> = action_str
       @transaction.set_action(transaction, action)
 
-      result = function.()
-
-      resp = @transaction.finish(transaction)
+      try do
+        function.()
+      catch
```

Comment:
This is a fairly large block with no comments. Can we get a comment or two explaining what the intent is here?

Reply:
You're right, this block was duplicating code from the ErrorHandler, because that implementation takes care of closing the Transaction, and expects a `%Plug.Conn{}` (which we don't have for channels).

I've created a new function in the ErrorHandler called `set_error/3`, which doesn't take a conn and just sets the error, so we move most of the error unpacking logic there.